multi-node/aws/pkg/config/templates/cluster.yaml

```diff
@@ -24,6 +24,10 @@ externalDNSName: {{.ExternalDNSName}}
 # it for you.
 #hostedZone: ""
 
+# The ID of hosted zone to add the externalDNSName to.
+# Only necessary if there are multiple hosted zones with the same name
+#hostedZoneId: ""
```

Comment:
It's not hard to find find the hostedZoneID, it's just very easy to remember the name.  It's nice to not have to go digging around in console or looking up aws-cli commands to use this feature.  It seems like requiring hostedZoneID adds a decent amount of friction.  Maybe we could list hosted zones and check if it's ambiguous as part of the validations in `kube-aws up` and then tell people they need to use the ID?

The wording on this seems to imply that hostedZoneID is only usable to disambiguate the hostedZoneName, but you could use hostedZoneID exclusively if you wanted.  Not sure if explaining that is worth the complexity it adds to figuring out how to configure cluster.yaml.

If we don't change the scope of this PR, the code LGTM.